modin/experimental/core/storage_formats/hdk/query_compiler.py

```diff
@@ -557,18 +553,27 @@ def concat(self, axis, other, **kwargs):
 
     def drop(self, index=None, columns=None, errors: str = "raise"):
         if index is not None:
-            raise NotImplementedError("Row drop")
+            # Only column drop is supported by the HDK engine
+            raise NotImplementedError("Index drop")
         if errors != "raise":
             raise NotImplementedError(
                 "This lazy query compiler will always "
                 + "raise an error on invalid columns."
             )
-        return self.__constructor__(
-            self._modin_frame.take_2d_labels_or_positional(
-                row_labels=index, col_labels=self.columns.drop(columns)
-            )
+
+        columns = self.columns.drop(columns)
+        new_frame = self._modin_frame.take_2d_labels_or_positional(
+            row_labels=index, col_labels=columns
         )
 
+        # If all columns are dropped and the index is trivial, we are
+        # not able to restore it, since we don't know the number of rows.
+        # In this case, we copy the index from the current frame.
+        if len(columns) == 0 and new_frame._index_cols is None:
+            new_frame._index_cache = self._modin_frame.index.copy()
```

Comment:
I suppose this is the behavior of pandas to save the index if there are no more columns? I wonder how this moment is handled by other backends.